lib/teleterm/daemon/daemon.go

```diff
@@ -1074,6 +1076,25 @@ func (s *Service) UpdateUserPreferences(ctx context.Context, clusterURI uri.Reso
 	return preferences, trace.Wrap(err)
 }
 
+// AuthenticateWebDevice is used to upgrade a web session (identified by a DeviceWebToken) to include device trust extensions.
+func (s *Service) AuthenticateWebDevice(ctx context.Context, rootClusterURI uri.ResourceURI, req *api.AuthenticateWebDeviceRequest) (*api.AuthenticateWebDeviceResponse, error) {
+	proxyClient, err := s.GetCachedClient(ctx, rootClusterURI)
```

Comment:
Personally I would write all of this in the base RPC handler (aka Handler. AuthenticateWebDevice) and do away with the changes here.

Reply:
We talked about this during the offsite. I had to add a new RPC and I was bemoaning our gRPC setup and Grzegorz said that you mentioned that as well in this PR.

I believe lib/teleterm was initially designed this way in order to decouple gRPC handlers from functions that perform the actual business logic (see https://github.com/gravitational/teleport/issues/14011#issuecomment-1195651304). At the time I had no experience working with gRPC, so I just stuck to the established convention.

Decoupling layers like that makes sense for HTTP routing, but nowadays I'm pretty sure it's unnecessary for gRPC. With HTTP, you have little control over what clients send to you, so it makes sense to have a layer dedicated to transforming an HTTP request into a valid message understood by your system. But with gRPC we already have protobufs, so this seems not needed. The messages sent with RCPs can work as inputs to your system. That's the "RPC" part of gRPC.